libraries/AP_Math/tests/test_math.cpp

```diff
@@ -0,0 +1,309 @@
+#include <AP_gtest.h>
+#include <AP_Math/AP_Math.h>
+
+#define SQRT_2 1.4142135623730951f
+
+
+TEST(ZeroTest, zero)
+{
+  #define TEST_IS_ZERO(_test, _result) { \
+    bool bZero = is_zero(_test); \
```

Comment:
No the compiler can figure this out himself. 
Only if people mix types you need to help sometimes.